pedpy/methods/profile_calculator.py

```diff
@@ -383,57 +472,86 @@ def compute_speed_profile(
     speed_method: SpeedMethod,
     grid_intersections_area: Optional[npt.NDArray[np.float64]] = None,
     fill_value: float = np.nan,
+    gaussian_width: float = 0.5,
+    # pylint: disable=too-many-arguments
 ) -> List[npt.NDArray[np.float64]]:
-    """Compute the speed profile.
+    """Computes the speed profile for pedestrians within an area.
+
+    This function calculates speed profiles based on pedestrian speed data
+    across a grid within a walkable area.
+    The method of computation can be selected among several options,
+    including mean (:attr:`SpeedMethod.MEAN`),
+    Gaussian (:attr:`SpeedMethod.GAUSSIAN`),
+    Voronoi (:attr:`SpeedMethod.VORONOI`),
+    and arithmetic mean methods (:attr:`SpeedMethod.ARITHMETIC`),
+    each suitable for different analysis contexts.
 
     Args:
-        data: Data from which the profiles are computes.
-            The DataFrame must contain a `frame` and a `speed` (result from
-            :func:`~speed_calculator.compute_individual_speed`) column.
-            For computing density profiles, it must contain a `polygon` column
+        data: A pandas DataFrame containing `frame` and pedestrian `speed`
+            (result from :func:`~speed_calculator.compute_individual_speed`).
+            Depending on `speed_method`, additional columns `x`, `y`, or
+            `polygon` might be required. `polygon` column
             (from :func:`~method_utils.compute_individual_voronoi_polygons`)
-            when using the `DensityMethod.VORONOI`. When computing the classic
-            density profile (`DensityMethod.CLASSIC`) the DataFrame needs to
-            contain the columns 'x' and 'y'. Computing the speed
-            profiles needs a `polygon` column (from
-            :func:`~method_utils.compute_individual_voronoi_polygons`) when
-            using the `SpeedMethod.VORONOI` or `SpeedMethod.ARITHMETIC`.
+            is required when using the :attr:`SpeedMethod.VORONOI` or
+            :attr:`SpeedMethod.ARITHMETIC`.
+            When computing the Gaussian profile (:attr:`SpeedMethod.GAUSSIAN`)
+            the DataFrame needs to contain the columns `x` and `y`.
             For getting a DataFrame containing all the needed data, you can
-            merge the results of the different function on the 'id' and
-            'frame' columns (see :meth:`pandas.DataFrame.merge` and
+            merge the results of the different function on the `id` and
+            `frame` columns (see :func:`pandas.DataFrame.merge` and
```

Comment:
This needs to be `` :meth:`pandas.DataFrame.merge`  ``, otherwise the link will not be displayed correctly.